app/views/media_objects/_add_to_playlist.html.erb

```diff
@@ -224,34 +216,41 @@ Unless required by applicable law or agreed to in writing, software distributed
           mediaObjectTitle = playlistForm.dataset.title;
           mediaObjectId = playlistForm.dataset.moid;
           let timelineScopes = getTimelineScopes();
-          scopes = timelineScopes.scopes;
+          let scopes = timelineScopes.scopes;
           streamId = timelineScopes.streamId || sectionIds[canvasIndex];
+          currentSectionLabel = `${mediaObjectTitle || mediaObjectId} - ${streamId}`;
 
           // Fill in the current track and section titles and custom scope times
           if(scopes?.length > 0) {
             let sectionInfo = scopes.filter(s => s.tags.includes('current-section'));
-            let currentSectionLabel = mediaObjectTitle;
             if(sectionInfo?.length > 0) {
               currentSectionLabel = sectionInfo[0].label || currentSectionLabel;
-              $('#current-section-name').text(currentSectionLabel);
             }
 
             let trackInfo = scopes.filter(s => s.tags.includes('current-track'));
             if(trackInfo.length > 0) {
               activeTrack = trackInfo[0];
-              let trackName = `${currentSectionLabel} - ${activeTrack.label || streamId}`;
+              let trackName = activeTrack.tags.includes('current-section')
+                ? activeTrack.label || streamId
+                : `${currentSectionLabel} - ${activeTrack.label || streamId}`;
```

Comment:
There is kind of a lot going on here logically.  Would it make sense to break this down into multiple lines?